dev/build-tool-support/src/main/scala/com/lightbend/lagom/dev/ManifestInfo.scala

```diff
@@ -0,0 +1,235 @@
+          s"that you are using, e.g. if you use [$highestVersion] all other modules that are released together MUST be of the " +
+          "same version. Make sure you're using a compatible set of libraries." +
+          s"Possibly conflicting versions [$conflictingVersions] in libraries [$fullInfo]"
+        Some(message)
+      } else
+        None
+    }
+
+}
+
+object ManifestInfo {
+
+  private val ImplTitle = "Implementation-Title"
+  private val ImplVersion = "Implementation-Version"
+  private val ImplVendor = "Implementation-Vendor-Id"
+
+  private val BundleName = "Bundle-Name"
+  private val BundleVersion = "Bundle-Version"
+  private val BundleVendor = "Bundle-Vendor"
+
+  // TODO: reconsider this if we want to make it more generic
+  // instead we should collect the versions only when doing the check and ProjectFamily should define the vendor id
+  private val knownVendors = Set(
+    "com.typesafe.akka",
+    "com.lightbend.akka",
+    "Lightbend Inc.",
+    "Lightbend",
+    "com.lightbend.lagom",
+    "com.typesafe.play"
+  )
+
+  /**
+   * Comparable version information
+   */
+  final class Version(val version: String) extends Comparable[Version] {
+    private val (numbers: Array[Int], rest: String) = {
+      val numbers = new Array[Int](3)
+      val segments: Array[String] = version.split("[.-]")
+      var segmentPos = 0
+      var numbersPos = 0
+      while (numbersPos < 3) {
+        if (segmentPos < segments.length) try {
+          numbers(numbersPos) = segments(segmentPos).toInt
+          segmentPos += 1
+        } catch {
+          case e: NumberFormatException ⇒
+            // This means that we have a trailing part on the version string and
+            // less than 3 numbers, so we assume that this is a "newer" version
+            numbers(numbersPos) = Integer.MAX_VALUE
+        }
+        numbersPos += 1
+      }
+
+      val rest: String =
+        if (segmentPos >= segments.length) ""
+        else String.join("-", util.Arrays.asList(util.Arrays.copyOfRange(segments, segmentPos, segments.length): _*))
+
+      (numbers, rest)
+    }
+
+    override def compareTo(other: Version): Int = {
+      var diff = 0
+      diff = numbers(0) - other.numbers(0)
+      if (diff == 0) {
+        diff = numbers(1) - other.numbers(1)
+        if (diff == 0) {
+          diff = numbers(2) - other.numbers(2)
+          if (diff == 0) {
+            diff = rest.compareTo(other.rest)
+          }
+        }
+      }
+      diff
+    }
+
+    override def equals(o: Any): Boolean = o match {
+      case v: Version ⇒ compareTo(v) == 0
+      case _          ⇒ false
+    }
+
+    override def toString: String = version
+
+    override def hashCode(): Int = {
+      val state = Seq(numbers, rest, version)
+      state.map(_.hashCode()).foldLeft(0)((a, b) => 31 * a + b)
+    }
+  }
+}
+
+trait ProjectFamily {
+  def familyName: String
+  def libraries: immutable.Seq[String]
+}
```

Comment:
We could have a jar containing only this and let other projects provide an implementation. The project specific impl could be automatically generated via a plugin.